### PR TITLE
Update Golang to 1.21

### PR DESCRIPTION
**Pull request by spolti**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/kserve/rest-proxy
 
-go 1.19
+go 1.21
```
Comment (ckadner) [inline go.mod:3]: Did you run `go mod tidy -compat=1.21` to update the dependencies below?

Comment (spolti) [reply go.mod:3]: just go mod tidy

Comment (spolti) [reply go.mod:3]: Just did, nothing else changed.

```
❯ go mod tidy -compat=1.21
❯ git status
On branch go1.21                                                                                                                                               ─╯
nothing to commit, working tree clean
```

Comment (ckadner) [reply go.mod:3]: cool. thanks

**Review by ckadner — APPROVED** [review]
/lgtm

**Pull request merged**